docs/api/lib/index.js

```diff
@@ -0,0 +1,153 @@
+const _ = require('lodash')
+const joiToSwagger = require('joi-to-swagger')
+
+const errorDocsLib = require('docs/api/lib/error')
+const extendedDocs = require('docs/api/lib/extended')
+
+/**
+ * Get methods for swagger routes
+ */
+const documentedMethods = ['GET', 'PUT', 'PATCH', 'DELETE', 'POST']
+const getMethod = methods => _.find(methods, method => _.includes(documentedMethods, method))
+
+/**
+ * Format parameters in query, params, body, header for docs
+ */
+const getFormattedValidationParams = (docsInKey, requiredKeys) => (obj, key) => {
+  const { description, ...schema } = obj
+  const isRequired = _.includes(requiredKeys, key)
+
+  return {
+    description,
+    in: docsInKey,
+    name: key,
+    required: isRequired,
+    schema,
+  }
+}
+
+function validateSection (objSchema, section) {
+  if (_.isEmpty(objSchema)) {
+    return []
+  }
+
+  const { swagger } = joiToSwagger(objSchema)
+  return _.map(swagger.properties, getFormattedValidationParams(section, swagger.required))
+}
```

Comment:
I would find this somewhat less contrived:

```suggestion
function validateSection (objSchema, section) {
  if (_.isEmpty(objSchema)) {
    return []
  }

  const { swagger } = joiToSwagger(objSchema)
  const requiredNamesSet = new Set(swagger.required)
  
  return _.map(swagger.properties, ({description, ...schema}, name) => ({
    description,
    in: section,
    name,
    required: requiredNamesSet.has(name),
    schema,
  }))
}
```